### PR TITLE
feat(web): controlled component behavior for DateRange

### LINKED ISSUE #1368: DateRange component doesn't function as a Controlled Component
**Affected Projects**
React

**Library Version:** x.y.z
3.4.1

**Describe the bug**
ReactiveSearch controlled DateRange component does not update properly on change. 

**To Reproduce**
https://codesandbox.io/s/ecstatic-ride-bly6r?fontsize=14&hidenavigation=1&theme=dark

**Expected behavior**
onChange should pass the correct params, and value should update the value appropriately 

**Desktop (please complete the following information):**
 - Browser: Brave/Chrome
 - 1.3.118 Chromium: 80.0.3987.116 (Official Build) (64-bit)

**Additional Information**
Extending the component doesn't seem to work either. Changing the value on the component doesn't trigger `beforeValueChange`, `onValueChange`, or `onQueryChange `

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
fixes #1368 

refactor controlled component behavior for DateRange

## How to test
- checkout `date-range-controlled` branch
- run `yarn && yarn dev:web`
- run `yarn start` inside `packages/web/examples/DateRangeControlled`

## Demo video - v1
https://www.loom.com/share/2343c77e69cc4be79f58b038ef6b205c

### CODE REVIEW

File: packages/web/examples/DateRangeControlled/src/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+				credentials="X8RsOu0Lp:9b4fe1a4-58c6-4089-a042-505d86d9da30"
+				type="listing"
+			>
+				<div className="row">
+					<div className="col">
+						<DateRange
+							value={this.state.date}
+							onChange={(value) => {
+								this.setState({
+									date: {
```
Comment (bietkul) [inline packages/web/examples/DateRangeControlled/src/index.js:58]: @ShahAnuj2610 I don't think that this example is useful or test anything. You're setting the same value which is already in the current state of `DateRange`.

File: packages/web/src/components/date/DateRange.js [hunk 2]
```diff
@@ -103,16 +103,21 @@ class DateRange extends Component {
...
-					true,
-					this.props,
-				);
+				const modDate = selectedValue
+					? {
+						start: this.props.selectedValue[0] || '',
+						end: this.props.selectedValue[1] || '',
+					} // prettier-ignore
+					: { start: '', end: '' };
+				if (value === undefined || (value && value.start === '' && value.end === '')) {
```
Comment (bietkul) [inline packages/web/src/components/date/DateRange.js]: @ShahAnuj2610 That seems wrong to me, why you're putting it under the `selectedValue` condition. Whenever the user changes the value it should update. Check the controlled behavior implementation in other components.

Comment (bietkul) [inline packages/web/src/components/date/DateRange.js]: @ShahAnuj2610 That seems wrong to me, why you're putting it under the `selectedValue` condition. Whenever user changes the value it should update. Check the controlled behavior implementation in other components.

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 Link the PR for the docs too.

File: packages/web/src/components/date/DateRange.js (hunk 2 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/date/DateRange.js]: @bietkul check this impl'n in MultiList https://github.com/appbaseio/reactivesearch/blob/bb62c348e2d623c91e3e448384d5b3f34b1b8a91/packages/web/src/components/list/MultiList.js#L170. It's the same.

**Comment by bietkul** [comment]
@ShahAnuj2610 Any update here?

**Comment by ShahAnuj2610** [comment]
> @ShahAnuj2610 Any update here?

> I don't think that this example is useful or test anything. You're setting the same value which is already in the current state of DateRange

@bietkul For other components, we show `controlled` component behavior similarly. Check https://github.com/appbaseio/reactivesearch/tree/3fe702ad62fd221fcaa31561f19424975f98c811/packages/web/examples/DataSearchControlled.

 ---

- Docs have `controlled component` behavior mentioned.

File: packages/web/examples/DateRangeControlled/src/index.js (hunk 1 above)
Comment (ShahAnuj2610) [reply packages/web/examples/DateRangeControlled/src/index.js:58]: @bietkul have updated the example

**Review by bietkul — APPROVED** [review]

**Pull request merged**